File: app/Models/Product.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Astrotomic\Translatable\Contracts\Translatable as TranslatableContract;
use Astrotomic\Translatable\Translatable;
use Illuminate\Database\Eloquent\SoftDeletes;

class Product extends Model implements TranslatableContract
{
    use Translatable;
    use HasFactory;
    use SoftDeletes;

    public $translatedAttributes = [
        'description'
    ];

    protected $fillable = [
        'name',
        'category_id',
        'brand_id',
        'price',
        'price_sale',
        'image',
        'stock_quantity'
    ];

    public function getUrlImgAttribute()
    {
        if (isset($this->image) && strpos($this->image, '/img/product') === 0) {
            return  asset("assets_home") . $this->image;
        } else {
            return asset('images') . '/' . $this->image;
        }
    }

    public function category()
    {
        return $this->hasone(Category::class, 'id', 'category_id');
    }
}

File: flask-app/product.py
class Product:
    def __init__(self, *args):
        self.id = args[0]
        self.name = args[1]
        self.category_id = args[2]
        self.supplier_id = args[3]
        self.price = args[4]
        self.discount = args[5]
        self.image = args[6]
        self.stock = args[7]
        self.created_at = args[8]
        self.updated_at = args[9]

    def product_to_dict(self):
        return {
            'id': self.id,
            'name': self.name,
            'category_id': self.category_id,
            'supplier_id': self.supplier_id,
            'price': self.price,
            'discount': self.discount,
            'image': self.image,
            'stock': self.stock,
            'created_at': self.created_at,
            'updated_at': self.updated_at
        }
